src/handle/compile.ml

```diff
@@ -104,40 +104,39 @@ let compile_file : string -> Sign.t = fun fname ->
   (* Run compilation. *)
   compile Stdlib.(!recompile) mp
 
-open Console
-
-(** Pure wrappers around compilation functions. Functions provided perform the
-    same computations as the ones defined earlier, but restores the state when
-    they have finished. An optional library mapping or state can be passed as
-    argument to change the settings. *)
-module Pure : sig
-  val compile : ?lm:Path.t*string -> ?st:State.t -> bool -> Path.t -> Sign.t
-  val compile_file : ?lm:Path.t*string -> ?st:State.t -> string -> Sign.t
-end = struct
-
-  (* [pure_apply_cfg ?lm ?st f] is function [f] but pure (without side
-     effects). The side effects taken into account occur in {!val:State.t} and
-     {!val:Library.lib_mappings}. Arguments [?lm] allows to set the library
-     mappings and [?st] sets the state. *)
-  let pure_apply_cfg :
-        ?lm:Path.t*string -> ?st:State.t -> ('a -> 'b) -> 'a -> 'b =
-    fun ?lm ?st f x ->
-    let libmap = !lib_mappings in
-    State.push ();
-    Option.iter Library.add_mapping lm;
-    Option.iter State.apply st;
-    let restore () =
-      State.pop ();
-      lib_mappings := libmap
-    in
-    try
-      let res = f x in
-      restore (); res
-    with e -> restore (); raise e
+(** [pure_apply_cfg ?lm ?st f] is function [f] but pure (without side
+   effects). The side effects taken into account occur in
+   {!val:Console.State.t} and {!val:Library.lib_mappings}. Arguments [?lm]
+   allows to set the library mappings and [?st] sets the state. *)
+let pure_apply_cfg :
+  ?lm:Path.t*string -> ?st:Console.State.t -> ('a -> 'b) -> 'a -> 'b =
+  fun ?lm ?st f x ->
+  let libmap = !lib_mappings
+  and unif_rules = !(Unif_rule.equiv.sym_rules)
+  and unif_dtree = !(Unif_rule.equiv.sym_dtree) in
+  Console.State.push ();
+  Option.iter Library.add_mapping lm;
+  Option.iter Console.State.apply st;
+  let restore () =
+    lib_mappings := libmap;
+    Unif_rule.equiv.sym_rules := unif_rules;
+    Unif_rule.equiv.sym_dtree := unif_dtree;
+    Console.State.pop ()
+  in
+  try let res = f x in restore (); res
+  with e -> restore (); raise e
 
-  let compile ?lm ?st force mp =
+(** /!\ For testing only. Pure wrappers around compilation
```

Comment:
It's not for testing only: programs that use lambdapi through its API often want to compile potentially unrelated signatures.

Reply:
I see. Could you please provide me with some examples?
As said in a previous comment, I removed Compile.Pure to avoid confusion with Pure (in src/pure/ directory).
It is annoying to have two different interfaces.
We should clarify this.
There is currently no documentation (except the source code) explaining which one to use.

Reply:
https://github.com/Deducteam/personoj/blob/c731827d3bdf238b7a70f5868f087c10eb846662/psnj_toolbox/bin/meta.ml#L14

Reply:
IIRC, there was some comment by Emilio that said that ideally the `Pure` as in `src/pure` should be merged with the main code.

Reply:
Sorry, I didn't know that you were using Compile.Pure, but this is easy to fix in your code.

Reply:
Comment fixed in d903e61c.